### DISCUSSION TITLE
Disable player moving by dragging

### DISCUSSION THREAD

**Discussion by cartermos** [mpvnet-player/mpv.net#672] [Q&A] [answered]
How can I disable moving the player by dragging the mouse?
Moving by the top border is fine, as it works in all other programs by default, that's all I need. But this dragging from the player window itself often leads to false movements of the player.

I tried to put **--no-window-dragging** in both mpv.conf and mpvnet.conf, but it doesn't work. Except that when I put it in mpv.conf, then the LMB click stops working in the program at all, but moving the player by dragging it still works, lol

I'm using v6.0.4.0 from Microsoft Store

**Comment by cartermos** [comment]
?

**Comment by Sneakpeakcss** [comment, marked as answer]
From what i remember version `6.0.4.0` didn't support `window-dragging` command yet, so that might be why, and newest versions are no longer released on MS Store:

[[1](https://github.com/mpvnet-player/mpv.net/blob/main/docs/changelog.md#v7003-beta-2023-12-15)]
> mpv.net can no longer be downloaded from the Microsoft store due to a general very poor experience with the package creation and submission. I've submitted mpv.net to the winget package repository, it's not yet processed.

So your best bet is to download [newest](https://github.com/mpvnet-player/mpv.net/releases/tag/v7.1.1.0) release and putting `window-dragging=no` in `mpv.conf`. 

Also keep in mind that jump to `v7` had some big changes and few things were unfortunately removed, so just in case backup your `mpv.net` config files before trying.

**Reply by cartermos** [reply]
So can you confirm that version 7 supports `window-dragging=no`? And what exactly was removed in version 7?

**Reply by Sneakpeakcss** [reply]
It 100% works, since i use it to disable window dragging in `uosc` script when a menu is visible, which wasn't previously possible. Initial [changelog](https://github.com/mpvnet-player/mpv.net/blob/main/docs/changelog.md#v7001-beta-2023-12-11) for `v7` has some information about what was removed, and there's also this:

https://github.com/mpvnet-player/mpv.net/blob/3f52c1255e68389c57b2ebdeb5f9bce09cdbf827/src/MpvNet.Windows/GuiCommand.cs#L387-L421

In some cases you'll get a notification that mentions that a function was removed and links to a lua script alternative. 

Personally the only thing that i've used and went missing is the command palette. Initially `playlist-random` was also removed, but that was reverted for compatibility sake since the alternative script had some issues with consecutive presses (*though at some point i found a fix for that one, so it might be axed again in the future if stax76 is interested in removing it*)

The mentioned changelog is more or less exhaustive when it comes to listing what was changed in general between `v6` and `v7`.

**Reply by cartermos** [reply]
Thanks, I've already downloaded the latest portable version and checked, finally `window-dragging=no` works.

And all my previous settings seem to work on the new version, only the `remember-window-position` doesn't work for some reason (when I start the player, it's in the middle again). Do you know what could be wrong?

**Reply by Sneakpeakcss** [reply]
Try checking if `remember-window-position=yes` is in correct file (`mpvnet.conf`) or generate a new `settings.xml` by removing the old one, other than that nothing else comes to mind.

**Reply by cartermos** [reply]
It didn't help. It's in the correct file. I additionally checked this by enabling and disabling this option in the new version of the program and the value appeared and disappeared in the `mpvnet.conf` file accordingly. So I think it is in the right place. I also deleted the `settings.xml` file, but a new one is created with the same content. Also in `mpvnet.conf` I have `start-size = always` (only these two values in total) and it works.

**Comment by cartermos** [comment]
> Try checking if `remember-window-position=yes` is in correct file (`mpvnet.conf`) or generate a new `settings.xml` by removing the old one, other than that nothing else comes to mind.

I found a solution to the problem of incorrect positioning somewhat unexpectedly - by changing the contents of the `<WindowLocation>` tag, or rather its internal `<X>` tag (which had a value of -1). If I set any other value there, then everything starts working and the player position is in the right place. I don't quite understand what the difference is between this tag and `<WindowPosition>`, which actually changes the position according to the specified values. Because `<WindowLocation>` doesn't seem to change much by itself, no matter what value I enter there - the player is moved by a couple of pixels at most. But with a value other than -1, it starts to show the position correctly, according to the values specified in `<WindowPosition>`. Anyway, I'm glad it helped me fix it.
